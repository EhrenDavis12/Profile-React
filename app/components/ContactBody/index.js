/**
 *
 * ContactBody
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';
import TextInputSingleLine from '../SharedKernel/InputBoxes/TextInputSingleLine';
import MessageInput from '../SharedKernel/InputBoxes/MessageInput';
import SubmitButton from '../SharedKernel/Buttons/SubmitButton';

function ContactBody() {
  return (
    <div className="container">
      <div className="row col-lg-12">
        <a href="https://www.linkedin.com/in/ehren-davis-ba078a2a/">
          To Indeed Profile
        </a>
      </div>
      <div className="email_form">
        <div className="row">
          <div className="col-md-6">
            <TextInputSingleLine
              label="Your name"
              placeholder="John Wick"
              id="fullName"
            />
          </div>
        </div>
        <div className="row col-12">
          <TextInputSingleLine
            label="Email address"
            placeholder="example@Gmail.com"
            id="emailAddress"
          />
        </div>
        <div className="row col-12">
          <MessageInput
            label="Please leave a message"
            id="email_message"
            rows="4"
          />
        </div>
        <div className="row col-12">
          <SubmitButton textValue="Submit" />
        </div>
      </div>
    </div>
  );
}

ContactBody.propTypes = {};

export default ContactBody;
