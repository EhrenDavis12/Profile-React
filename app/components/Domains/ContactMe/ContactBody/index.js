/**
 *
 * ContactBody
 *
 */

import React from 'react';

import PropTypes from 'prop-types';
import './styles.css';
// import styled from 'styled-components';
import BasicButton from 'components/Domains/SharedKernel/Buttons/BasicButton';
import validator from 'email-validator';
import classNames from 'classnames';

// function ContactBody() {
class ContactBody extends React.Component {
  state = {};

  submit = () => {
    const email = this.clientEmailField.value;
    if (!validator.validate(email)) {
      this.setState({ errorText: 'Please provide a valid email' });
      return;
    }
    this.setState({ errorText: null });

    this.props.submitMessageForm({
      contactInfo: email,
      userUuid: '2f67b460-6469-11e9-b132-2b223266dc25',
      subject: this.subjectField.value,
      message: this.messageField.value,
    });
  };

  clear = () => {
    this.setState({ errorText: '' });
    this.clientEmailField.value = '';
    this.subjectField.value = '';
  };

  render() {
    const fieldError = this.state.errorText ? (
      <div className="errorMessage">{this.state.errorText}</div>
    ) : null;
    return (
      <div className="messageForm">
        <input
          className={classNames('input', {
            inputError: this.state.errorText,
          })}
          placeholder="Your Email"
          ref={f => {
            this.clientEmailField = f;
          }}
          type="text"
        />
        {fieldError}
        <input
          className="input"
          placeholder="Subject"
          ref={f => {
            this.subjectField = f;
          }}
          type="text"
        />
        <textarea
          className="inputLargeText"
          placeholder="Your Message"
          ref={f => {
            this.messageField = f;
          }}
          type="text"
        />
        <div className="actionContainer">
          <BasicButton className="button" onClick={this.clear}>
            Clear
          </BasicButton>
          <BasicButton className="button" onClick={this.submit}>
            Submit
          </BasicButton>
        </div>
      </div>
    );
  }
}

ContactBody.propTypes = {
  // responseTest: PropTypes.arrayOf(PropTypes.shape({})),
  /* responseTest: PropTypes.shape({
    data: PropTypes.shape({
      auth0Id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired, */
  submitMessageForm: PropTypes.func.isRequired,
};

export default ContactBody;
