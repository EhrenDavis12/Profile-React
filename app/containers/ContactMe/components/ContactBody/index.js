/**
 *
 * ContactBody
 *
 */

import React from 'react';

import PropTypes from 'prop-types';
import './styles.css';
// import styled from 'styled-components';
import BasicButton from 'components/Buttons/BasicButton';
import validator from 'email-validator';
import classNames from 'classnames';

// function ContactBody() {
class ContactBody extends React.Component {
  state = {};

  submit = () => {
    const email = this.clientEmailField.value;
    if (!validator.validate(email)) {
      this.setState({ errorText: 'Please provide a valid email' });
      return;
    }
    this.setState({ errorText: null });

    this.props.submitMessageForm({
      contactInfo: email,
      userUuid: process.env.USER_UUID,
      subject: this.subjectField.value,
      message: this.messageField.value,
    });
  };

  clear = () => {
    this.setState({ errorText: '' });
    this.messageField.value = '';
    this.clientEmailField.value = '';
    this.subjectField.value = '';
  };

  render() {
    const fieldError = this.state.errorText ? (
      <div className="errorMessage">{this.state.errorText}</div>
    ) : null;
    return (
      <div className="messageForm">
        <input
          id="emailInput"
          className={classNames('input', {
            inputError: this.state.errorText,
          })}
          placeholder="Your Email"
          ref={f => {
            this.clientEmailField = f;
          }}
          type="email"
        />
        {fieldError}
        <input
          id="subjectInput"
          className="input"
          placeholder="Subject"
          ref={f => {
            this.subjectField = f;
          }}
          type="text"
        />
        <textarea
          id="messageInput"
          className="inputLargeText"
          placeholder="Your Message"
          ref={f => {
            this.messageField = f;
          }}
          type="text"
        />
        <div className="actionContainer">
          <BasicButton id="clear" className="button" onClick={this.clear}>
            Clear
          </BasicButton>
          <BasicButton id="submit" className="button" onClick={this.submit}>
            Submit
          </BasicButton>
        </div>
      </div>
    );
  }
}

ContactBody.propTypes = {
  // responseTest: PropTypes.arrayOf(PropTypes.shape({})),
  /* responseTest: PropTypes.shape({
    data: PropTypes.shape({
      auth0Id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired, */
  submitMessageForm: PropTypes.func.isRequired,
};

export default ContactBody;
