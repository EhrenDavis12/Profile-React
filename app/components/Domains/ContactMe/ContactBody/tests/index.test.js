import React from 'react';
import renderer from 'react-test-renderer';
// import { mount } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

import ContactBody from '../index';

/* describe('<ContactBody />', () => {
  it('Expect to have unit tests specified', () => {
    expect(true).toEqual(false);
  });
}); */

describe('<ContactBody />', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(<ContactBody submitMessageForm={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
