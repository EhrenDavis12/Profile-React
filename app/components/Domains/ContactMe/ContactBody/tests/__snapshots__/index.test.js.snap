// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactBody /> renders as expected 1`] = `
<div
  className="messageForm"
>
  <input
    className="input"
    placeholder="Your Email"
    type="text"
  />
  <input
    className="input"
    placeholder="Subject"
    type="text"
  />
  <textarea
    className="inputLargeText"
    placeholder="Your Message"
    type="text"
  />
  <div
    className="actionContainer"
  >
    <div
      className="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Clear
    </div>
    <div
      className="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Submit
    </div>
  </div>
</div>
`;

exports[`ContactBody renders as expected 1`] = `
<div
  className="messageForm"
>
  <input
    className="input"
    placeholder="Your Email"
    type="text"
  />
  <input
    className="input"
    placeholder="Subject"
    type="text"
  />
  <textarea
    className="inputLargeText"
    placeholder="Your Message"
    type="text"
  />
  <div
    className="actionContainer"
  >
    <div
      className="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Clear
    </div>
    <div
      className="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Submit
    </div>
  </div>
</div>
`;
